src/parseTools.js

```diff
@@ -1441,8 +1441,32 @@ function makeEval(code) {
 }
 
 function makeStaticAlloc(size) {
-  size = (size + (STACK_ALIGN-1)) & -STACK_ALIGN;
-  return 'STATICTOP; STATICTOP += ' + size + ';';
+  size = alignMemory(size);
+  var ret = alignMemory(GLOBAL_BASE + STATIC_BUMP);
+  STATIC_BUMP = ret + size - GLOBAL_BASE;
+  return ret;
+}
+
+function makeStaticString(string) {
+  var len = string.length + 1;
```

Comment:
This is not correct size computation for an UTF-8 string - either need to use `lengthBytesUTF8(string)` or if you want to conservatively allocate, use `string.length*4+1`.

Reply:
Thanks! Fixing with lengthBytesUTF8 at compile time, to not waste any space here.